### PR TITLE
Look the process, not only payment settings

### PR DESCRIPTION
**Pull request by rap1ds**
- Commission counter under new listing price
- Payment settings registeration guard before posting new listing
- Warning about open listings but no payment info

### CODE REVIEW

File: app/controllers/listings_controller.rb [hunk 1]
```diff
@@ -117,7 +117,7 @@ def show
     # TODO Change this so that the path is always the same, but the controller
     # decides what to do. We don't want to make a API call to TransactionService
     # just to show a listing details
-    process = get_transaction_process(community: @current_community, listing: @listing)
+    process = get_transaction_process(community: @current_community, transaction_process_id: @listing.transaction_process_id)
```
Comment (sktoiva) [inline app/controllers/listings_controller.rb]: Why don't just pass the community_id? 

The signature would be then uniform.

File: app/controllers/listings_controller.rb [hunk 2]
```diff
@@ -360,19 +363,19 @@ def build_title(params)
...
-    when nil
+    case [payment_type, process]
+    when matches([__, :none])
       {seller_commission_in_use: false,
-       payment_gateway: payment_type,
+       payment_gateway: nil,
        minimum_commission: Money.new(0, currency),
        commission_from_seller: 0,
        minimum_price_cents: 0}
-    when :paypal
```
Comment (sktoiva) [inline app/controllers/listings_controller.rb]: shouldn't it be `[__, :paypal]?`

File: app/controllers/listings_controller.rb [hunk 3]
```diff
@@ -360,19 +363,19 @@ def build_title(params)
...
+    when matches([:paypal])
```
Comment (sktoiva) [inline app/controllers/listings_controller.rb]: shouldn't it be [__, :paypal]?

File: app/view_utils/paypal_helper.rb [hunk 4]
```diff
@@ -85,7 +85,19 @@ def paypal_provisioned?(community_id)
   def open_listings_with_missing_payment_info?(user_id, community_id)
     paypal_active?(community_id) &&
     !user_and_community_ready_for_payments?(user_id, community_id) &&
-    !MarketplaceService::Listing::Query.open_listings_with_price_for(community_id, user_id).empty?
+    open_listings_with_price_and_payment_process?(community_id, user_id)
+  end
+
+  def open_listings_with_price_and_payment_process?(community_id, user_id)
+    listings = MarketplaceService::Listing::Query.open_listings_with_price_for(community_id, user_id)
```
Comment (ovan) [inline app/view_utils/paypal_helper.rb]: Get rid of the price check, it's old logic. Instead get all unique process ids for users listings in this community and iterate through processes to see if any of them requires payments.

**Pull request merged**